syft/pkg/catalog.go

```diff
@@ -38,21 +41,26 @@ func (c *Catalog) PackageCount() int {
 }
 
 // Package returns the package with the given ID.
-func (c *Catalog) Package(id ID) *Package {
+func (c *Catalog) Package(id artifact.ID) *Package {
```

Comment:
Do we return a copy now because we want to force the immutability of the catalog?

Reply:
exactly 👍 I just updated the PR description a minute ago to reflect these thoughts